pkg/query-service/app/logs/v3/query_builder.go

```diff
@@ -113,7 +110,8 @@ func getSelectLabels(aggregatorOperator v3.AggregateOperator, groupBy []v3.Attri
 		selectLabels = ""
 	} else {
 		for _, tag := range groupBy {
-			columnName, err := getClickhouseColumnName(tag, fields)
+			enrichedTag := enrichFieldWithMetadata(tag, fields)
```

Comment:
Do you need this since `getClickhouseColumnName` is internally calling `enrichFieldWithMetadata`?

Reply:
The getClickhouseColumnName function is updated, it doesn't call the  enrichFieldWithMetadata.